### DISCUSSION TITLE
log4j 2 throws log4j 1 warnings

### DISCUSSION THREAD

**Discussion by Nerdy-5** [apache/logging-log4j2#2003] [Q&A]
## Description

In hierarchy.java, the lines

log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.

make it seem, like log4j 1.x is failing to boot up, providing a link to outdated documentation.

## Configuration

**Version:** [2.20.0]

**Operating system:** [AIX]

**JDK:** [OpenJDK 11]

## Logs

```
log4j:WARN No appenders could be found for logger (xxx).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
```

## Reproduction

n.a.

**Comment by ppkarwasz** [comment]
Messages starting with `log4j:WARN` are really generated by Log4j 1.x (or Reload4j, see qos-ch/reload4j#69).

I am moving this to discussions.

**Comment by Raphasha27** [comment]
This happens when you have log4j 1.x API jars on the classpath alongside log4j 2.x. The `log4j-1.2-api` bridge is being triggered but the old API classes are also present.

**Fix:** Find and remove old log4j 1.x jars, keeping only the 2.x bridge if needed:

```bash
# Find conflicts
mvn dependency:tree -Dincludes=log4j
```

Look for:
- `log4j:log4j:1.2.x` — remove this (old log4j 1.x)
- Keep `org.apache.logging.log4j:log4j-1.2-api:2.x` (the bridge, only if you have code using log4j 1.x API)

If you're using SLF4J, you don't need the bridge at all:

```xml
<dependency>
    <groupId>org.apache.logging.log4j</groupId>
    <artifactId>log4j-slf4j2-impl</artifactId>
    <version>2.20.0</version>
</dependency>
```

**Exclude the transitive log4j 1.x dependency:**
```xml
<dependency>
    <groupId>some.library</groupId>
    <artifactId>some-artifact</artifactId>
    <exclusions>
        <exclusion>
            <groupId>log4j</groupId>
            <artifactId>log4j</artifactId>
        </exclusion>
    </exclusions>
</dependency>
```

The warning is harmless but noisy — removing the old jar silences it entirely.
